src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java

```diff
@@ -0,0 +1,182 @@
+package com.mercuryCyclists.Sales.service;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.mercuryCyclists.Sales.entity.InStoreSale;
+import com.mercuryCyclists.Sales.entity.Store;
+import com.mercuryCyclists.Sales.repository.InStoreSaleRepository;
+import com.mercuryCyclists.Sales.repository.StoreRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class InStoreSaleService {
+
+    private final InStoreSaleRepository inStoreSaleRepository;
+    private final StoreRepository storeRepository;
+    private final SaleService saleService;
+
+    @Autowired
+    public InStoreSaleService(InStoreSaleRepository inStoreSaleRepository, StoreRepository storeRepository, SaleService saleService) {
+        this.inStoreSaleRepository = inStoreSaleRepository;
+        this.storeRepository = storeRepository;
+        this.saleService = saleService;
+    }
+
+    /**
+     * Gets all In Store Sales
+     */
+    public List<InStoreSale> getAllInStoreSales() {
+        return inStoreSaleRepository.findAll();
+    }
+
+    /**
+     * Gets In Store Sales by Store ID
+     */
+    public List<InStoreSale> getStoreSales(Long storeId) {
+        Optional<Store> store = storeRepository.findById(storeId);
+
+        if (!store.isPresent()) {
+            throw new IllegalStateException(String.format("Store with Id %s does not exist", storeId));
+        }
+
+        return inStoreSaleRepository.findInStoreSalesByStore(store.get());
+    }
+
+    /**
+     * Register a new In Store Sale
+     */
+    public ResponseEntity<InStoreSale> registerInStoreSale(InStoreSale inStoreSale, Long storeId) {
+        // Validate sale
+        if (!inStoreSale.validate()) {
+            throw new IllegalStateException("Invalid In Store Sale");
+        }
+
+        // Do after first validation in case quantity is null
+        if (inStoreSale.getQuantity() <= 0) {
+            throw new IllegalStateException("In Store Sale has 0 or a negative quantity");
+        }
```

Comment:
Have a look at zhifas check, for a simplified v.
If you dont want to add a dependency, i would suggest putting this in the obj validate method

Reply:
Just had a look, will add it is a dependency

Reply:
Dependency not found, just adding to validate method